WireUI/Sources/WireUIFoundation/Main/MainCoordinator.swift

```diff
@@ -0,0 +1,364 @@
+
+    public func showConversationList(conversationFilter: (some MainConversationFilterRepresentable)?) {
+        defer {
+            // switch to the conversation list tab
+            tabBarController.selectedContent = .conversations
+
+            // apply the filter to the conversation list
+            let mainConversationFilter = conversationFilter?.map()
+            conversationList.conversationFilter = mainConversationFilter.map { .init($0) }
+
+            // set the right menu item in the sidebar
+            switch mainConversationFilter {
+            case .none:
+                sidebar.selectedMenuItem = .init(.all)
+            case .favorites:
+                sidebar.selectedMenuItem = .init(.favorites)
+            case .groups:
+                sidebar.selectedMenuItem = .init(.groups)
+            case .oneOnOne:
+                sidebar.selectedMenuItem = .init(.oneOnOne)
+            }
+        }
+
+        // In collapsed state switching the tab was all we needed to do.
+        guard mainSplitViewState == .expanded else { return }
+
+        dismissArchiveIfNeeded()
+        dismissNewConversationIfNeeded()
+        dismissSettingsIfNeeded()
+        dismissSelfProfileIfNeeded()
+
+        // Move the conversation list from the tab bar controller to the split view controller if needed.
+        if let conversationList = tabBarController.conversations?.conversationList {
+            tabBarController.conversations = nil
+            splitViewController.conversationList = conversationList
+        }
+    }
+
+    public func showArchivedConversations() {
+        // switch to the archive tab
+        tabBarController.selectedContent = .archive
+
+        // In collapsed state switching the tab was all we needed to do.
+        guard mainSplitViewState == .expanded else { return }
+
+        dismissConversationListIfNeeded()
+        dismissNewConversationIfNeeded()
+        dismissSettingsIfNeeded()
+        dismissSelfProfileIfNeeded()
+
+        // move the archive from the tab bar controller to the split view controller
+        if let archive = tabBarController.archive {
+            tabBarController.archive = nil
+            splitViewController.archive = archive
+        }
+    }
+
+    public func showSettings() {
+        tabBarController.selectedContent = .settings
+
+        // In collapsed state switching the tab was all we needed to do.
+        guard mainSplitViewState == .expanded else { return }
+
+        dismissConversationListIfNeeded()
+        dismissArchiveIfNeeded()
+        dismissNewConversationIfNeeded()
+        dismissSelfProfileIfNeeded()
+
+        // move the settings from the tab bar controller to the split view controller
+        if let settings = tabBarController.settings {
+            tabBarController.settings = nil
+            splitViewController.settings = settings
+        }
+    }
+
+    public func showSelfProfile() {
+        guard selfProfile == nil else { return }
```

Comment:
Optional: I assume if `selfProfile` is not `nil` this method should not have been called. If that is true, consider logging error / asserting in this case and other similar cases.

Reply:
I wanted to log, however, the `WireLogger` types are in `WireSystem` which is an Xcode project and cannot be imported here.

Reply:
👍 Oh yeah. We should extract it soon then